src/apps/frontend/contexts/auth.provider.tsx

```diff
@@ -49,7 +49,11 @@ const loginFn = async (
 ): Promise<ApiResponse<AccessToken>> => {
   const result = await authService.login(username, password);
   if (result.data) {
-    localStorage.setItem('access-token', JSON.stringify(result.data));
+    localStorage.setItem('access-token', JSON.stringify({
```

Comment:
Serialization, and deserialization should be encapsulated into the object.

Consider creating `to_json` within `AccessToken` class to properly encapsulate the serialization so that the rest of the codebase does not need to know "how" to do serialization.